src/main/kotlin/com/dprint/formatter/DprintExternalFormatter.kt

```diff
@@ -24,10 +27,28 @@ class DprintExternalFormatter : AsyncDocumentFormattingService() {
     }
 
     override fun canFormat(file: PsiFile): Boolean {
-        val editorService = file.project.service<EditorServiceManager>().maybeGetEditorService()
-        return editorService != null && file.virtualFile != null &&
-            file.project.service<ProjectConfiguration>().state.enabled &&
-            editorService.canFormat(file.virtualFile.path)
+        try {
+            val future = CompletableFuture.supplyAsync {
+                val editorService = file.project.service<EditorServiceManager>().maybeGetEditorService()
+                editorService != null &&
+                    file.virtualFile != null &&
+                    file.project.service<ProjectConfiguration>().state.enabled &&
+                    editorService.canFormat(file.virtualFile.path)
```

Comment:
Yes that is correct. The issue is not that this part is being called twice but it is being called from both here and [also here](https://github.com/dprint/dprint-intellij/blob/main/src/main/kotlin/com/dprint/services/FormatterService.kt#L74). 

I wrapped the part that talks to the underlying process with `synchronize` blocks hoping that it would solve the issue and it did before the refactor for v5. I am not entirely sure if the issue was inside that or with something else in the method so I have now put a `@Synchronize` annotation on the method to hopefully ensure that it cannot be called at the same time on different threads. 

It seems to be working so far but the issues is really hard to reproduce and so far I have only accidentally hit it while working.

Reply:
Also one final note, I am pretty sure this is running on a single process and multiple threads. The IntelliJ docs are pretty bad so I am not 100% sure. If it is running on multiple processes this would definitely explain the issues I am seeing. I can't see why they would be taking the perf hit of spinning up multiple jvm's though.